prov/efa/test/efa_unit_test_common.c

```diff
@@ -180,3 +182,61 @@ void efa_unit_test_eager_msgrtm_pkt_construct(struct rxr_pkt_entry *pkt_entry, s
 	assert_int_equal(*connid, attr->connid);
 	pkt_entry->pkt_size = sizeof(base_hdr) + sizeof(opt_connid_hdr);
 }
+
+/**
+ * @brief Construct RXR_HANDSHAKE_PKT
+ *	The function will include the optional connid/host id headers if and only if
+ *	attr->connid/host id are non-zero.
+ *
+ * @param[in,out] pkt_entry Packet entry. Must be non-NULL.
+ * @param[in] attr Packet attributes.
+ */
+void efa_unit_test_handshake_pkt_construct(struct rxr_pkt_entry *pkt_entry, struct efa_unit_test_handshake_pkt_attr *attr)
+{
+
+	int nex = (RXR_NUM_EXTRA_FEATURE_OR_REQUEST - 1) / 64 + 1;
+	struct rxr_handshake_hdr *handshake_hdr = (struct rxr_handshake_hdr *)pkt_entry->wiredata;
+
+	handshake_hdr->type = RXR_HANDSHAKE_PKT;
+	handshake_hdr->version = RXR_PROTOCOL_VERSION;
+	handshake_hdr->nextra_p3 = nex + 3;
+	handshake_hdr->flags = 0;
+
+	calloc((uintptr_t)handshake_hdr->extra_info, nex * sizeof(uint64_t));
+	pkt_entry->pkt_size = sizeof(struct rxr_handshake_hdr) + nex * sizeof(uint64_t);
+	memcpy(pkt_entry->wiredata, handshake_hdr, sizeof(struct rxr_handshake_hdr));
+	assert_int_equal(rxr_get_base_hdr(pkt_entry->wiredata)->type, RXR_HANDSHAKE_PKT);
+
+	if (attr->connid) {
+		struct rxr_handshake_opt_connid_hdr opt_connid_hdr = {0};
+		opt_connid_hdr.connid = attr->connid;
+		handshake_hdr->flags |= RXR_PKT_CONNID_HDR;
+		memcpy(rxr_pkt_entry_wiredata_end(pkt_entry), &opt_connid_hdr, sizeof(struct rxr_handshake_opt_connid_hdr));
```

Comment:
I do not think use `rxr_pkt_entry_wiredata_end` is good idea. It indirectly refer to `pkt_entry->pkt_size`, which was updated on the fly.

Reply:
Addressed.